src/components/ContentHighlights/HighlightStepper/HighlightStepperTitleInput.jsx

```diff
@@ -11,37 +11,59 @@ const HighlightStepperTitleInput = () => {
   const highlightTitle = useContextSelector(ContentHighlightsContext, v => v[0].stepperModal.highlightTitle);
   const [titleLength, setTitleLength] = useState(highlightTitle?.length || 0);
   const [isInvalid, setIsInvalid] = useState(false);
-
-  const handleChange = (e) => {
+  const [highlightValue, setHighlightValue] = useState({
+    initialized: false,
+    highlightTitle: highlightTitle || '',
+    titleStepValidationError: undefined,
+    highlightTitleLength: titleLength || 0,
+  });
+  const handleChange = ((e) => {
+    e.persist();
     if (e.target.value.length > MAX_HIGHLIGHT_TITLE_LENGTH) {
       setIsInvalid(true);
-      setHighlightTitle({
+      setHighlightValue({
+        initialized: true,
         highlightTitle: e.target.value,
         titleStepValidationError: DEFAULT_ERROR_MESSAGE.EXCEEDS_HIGHLIGHT_TITLE_LENGTH,
+        highlightTitleLength: e.target.value.length,
       });
     } else {
       setIsInvalid(false);
-      setHighlightTitle({
+      setHighlightValue({
+        initialized: true,
         highlightTitle: e.target.value,
         titleStepValidationError: undefined,
+        highlightTitleLength: e.target.value.length,
       });
     }
-    setTitleLength(e.target.value.length);
-  };
+  });
 
+  useEffect(() => {
+    if (highlightValue.initialized) {
+      setHighlightTitle({
+        highlightTitle: highlightValue.highlightTitle,
+        titleStepValidationError: highlightValue.titleStepValidationError,
+      });
+      setTitleLength(highlightValue.highlightTitleLength);
+      setHighlightValue({
+        ...highlightValue,
+        initialized: false,
+      });
+    }
+  }, [highlightTitle, setHighlightTitle, highlightValue]);
   return (
     <Form.Group
       isInvalid={isInvalid}
     >
       <Form.Control
         data-testid="stepper-title-input"
-        value={highlightTitle || ''}
+        value={highlightValue.highlightTitle || ''}
```

Comment:
With your refactor above, `highlightTitle` already falls back to `''`, right?

Reply:
Correct, verified

Reply:
Haha, cool. The point here was that the `|| ''` fallback on this line is largely unnecessary because `highlightValue.highlightTitle` already defines the fallback elsewhere.